### ISSUE TITLE
Hex colors in gradients may not be shrinked properly (regressed since 4.1.5)

### ISSUE THREAD

**Issue by ngyikp** [jakubpawlowicz/clean-css#959]
Not a serious bug, but more of a degradation

* clean-css version - `npm ls clean-css`: 4.1.7, but regressed since 4.1.5, most likely https://github.com/jakubpawlowicz/clean-css/issues/945
* node.js version - `node -v`: v8.1.4
* operating system: macOS 10.12.5

### Configuration options

```js
const CleanCSS = require('clean-css');
const fs = require('fs');

const input = fs.readFileSync('test.css', 'utf-8');
console.log(new CleanCSS().minify(input).styles);
```

### Input CSS

```css
.a {background-image:linear-gradient(to top, #AABBCC, #ffffff);}
.b {background-image:linear-gradient(to top, #AABBCC, #FFFFFF);}
```

### Actual output CSS

```css
.a{background-image:linear-gradient(to top,#abc,#ffffff)}
.b{background-image:linear-gradient(to top,#abc,#fffFFF)}
```

### Expected output CSS

```css
.a,.b{background-image:linear-gradient(to top,#abc,#fff)}
```

Hex colors in gradients are failing to shrink and some weird capitalisation if they are uppercase.

**Comment by jakubpawlowicz** [comment]
This is weird indeed. Thanks for catching it!

**Comment by jakubpawlowicz** [comment]
Thanks @ngyikp, it's fixed in 4.1.8.
